### PR TITLE
Disable partial results for secondary queries

### LINKED ISSUE #55864: Distributed: Bad cast from type DB::ColumnVector<unsigned long> to DB::ColumnAggregateFunction.
Report 1: https://s3.amazonaws.com/clickhouse-test-reports/55813/297eda83fc153c2e83a9d8a65ffb9d6e12f52bbd/stress_test__msan_.html

Report 2: https://s3.amazonaws.com/clickhouse-test-reports/55860/01093d4698c3381bed7bf69505d6bb9d62360e76/upgrade_check__asan_.html

**Describe the bug**

```
2023.10.20 11:00:00.764584 [ 5806 ] {73f441d2-6a30-4dc0-bb3b-db2673d34eac} <Debug> executeQuery: (from [::ffff:127.0.0.1]:54654, initial_query_id: b4a2c158-c3f3-4469-9372-7837ba549e13) (comment: 01290_max_execution_speed_distributed.sql) SELECT count() FROM numbers(1000000) (stage: WithMergeableState)
2023.10.20 11:00:00.764928 [ 5806 ] {73f441d2-6a30-4dc0-bb3b-db2673d34eac} <Trace> InterpreterSelectQuery: FetchColumns -> WithMergeableState
2023.10.20 11:00:00.765169 [ 50749 ] {73f441d2-6a30-4dc0-bb3b-db2673d34eac} <Trace> AggregatingTransform: Aggregating
2023.10.20 11:00:00.765192 [ 50749 ] {73f441d2-6a30-4dc0-bb3b-db2673d34eac} <Trace> Aggregator: An entry for key=1965726198748367308 found in cache: sum_of_sizes=1, median_size=1
2023.10.20 11:00:00.765199 [ 50749 ] {73f441d2-6a30-4dc0-bb3b-db2673d34eac} <Trace> Aggregator: Aggregation method: without_key
2023.10.20 11:00:01.571225 [ 5806 ] {73f441d2-6a30-4dc0-bb3b-db2673d34eac} <Error> executeQuery: Code: 49. DB::Exception: Bad cast from type DB::ColumnVector<unsigned long> to DB::ColumnAggregateFunction. (LOGICAL_ERROR) (version 23.9.2.56 (official build)) (from [::ffff:127.0.0.1]:54654) (comment: 01290_max_execution_speed_distributed.sql) (in query: SELECT count() FROM numbers(1000000)), Stack trace (when copying this message, always include the lines below):
2023.10.20 11:00:01.572914 [ 5806 ] {73f441d2-6a30-4dc0-bb3b-db2673d34eac} <Error> TCPHandler: Code: 49. DB::Exception: Bad cast from type DB::ColumnVector<unsigned long> to DB::ColumnAggregateFunction. (LOGICAL_ERROR), Stack trace (when copying this message, always include the lines below):
```

The originating query is using remote over localhost to check speed limits on distributed queries:
```
2023.10.20 11:00:00.699311 [ 5057 ] {b4a2c158-c3f3-4469-9372-7837ba549e13} <Debug> executeQuery: (from [::1]:56026) (comment: 01290_max_execution_speed_distributed.sql) SELECT count('special query for 01290_max_execution_speed_distributed') FROM remote('127.0.0.{2,3}', numbers(1000000)); (stage: Complete)
2023.10.20 11:00:00.699588 [ 5057 ] {b4a2c158-c3f3-4469-9372-7837ba549e13} <Trace> ContextAccess (default): Access granted: CREATE TEMPORARY TABLE, REMOTE ON *.*
2023.10.20 11:00:00.699696 [ 5057 ] {b4a2c158-c3f3-4469-9372-7837ba549e13} <Trace> Connection (127.0.0.2:9000): Connecting. Database: (not specified). User: default
2023.10.20 11:00:00.760549 [ 5057 ] {b4a2c158-c3f3-4469-9372-7837ba549e13} <Trace> Connection (127.0.0.2:9000): Connected to ClickHouse server version 23.9.2.
2023.10.20 11:00:00.761184 [ 5682 ] {be6d282e-fe46-4f0d-81e0-9c24d172c95f} <Debug> executeQuery: (from [::ffff:127.0.0.1]:54668, initial_query_id: b4a2c158-c3f3-4469-9372-7837ba549e13) (comment: 01290_max_execution_speed_distributed.sql) DESC TABLE numbers(1000000) (stage: Complete)
2023.10.20 11:00:00.762203 [ 5057 ] {b4a2c158-c3f3-4469-9372-7837ba549e13} <Trace> InterpreterSelectQuery: WithMergeableState -> Complete
2023.10.20 11:00:00.763893 [ 154098 ] {b4a2c158-c3f3-4469-9372-7837ba549e13} <Trace> Connection (127.0.0.3:9000): Connecting. Database: (not specified). User: default
2023.10.20 11:00:00.764106 [ 50395 ] {b4a2c158-c3f3-4469-9372-7837ba549e13} <Debug> Connection (127.0.0.2:9000): Sent data for 2 scalars, total 2 rows in 3.337e-05 sec., 58447 rows/sec., 68.00 B (1.89 MiB/sec.), compressed 0.4594594594594595 times to 148.00 B (4.10 MiB/sec.)
2023.10.20 11:00:00.764234 [ 50395 ] {b4a2c158-c3f3-4469-9372-7837ba549e13} <Debug> Connection (127.0.0.2:9000): Sent data for 1 external tables, total 1 rows in 6.4094e-05 sec., 15481 rows/sec., 36.00 B (542.17 KiB/sec.), compressed 0.3673469387755102 times to 98.00 B (1.44 MiB/sec.)
2023.10.20 11:00:00.764584 [ 5806 ] {73f441d2-6a30-4dc0-bb3b-db2673d34eac} <Debug> executeQuery: (from [::ffff:127.0.0.1]:54654, initial_query_id: b4a2c158-c3f3-4469-9372-7837ba549e13) (comment: 01290_max_execution_speed_distributed.sql) SELECT count() FROM numbers(1000000) (stage: WithMergeableState)
2023.10.20 11:00:00.829208 [ 154098 ] {b4a2c158-c3f3-4469-9372-7837ba549e13} <Trace> Connection (127.0.0.3:9000): Connected to ClickHouse server version 23.9.2.
2023.10.20 11:00:00.829412 [ 154098 ] {b4a2c158-c3f3-4469-9372-7837ba549e13} <Debug> Connection (127.0.0.3:9000): Sent data for 2 scalars, total 2 rows in 3.5724e-05 sec., 55276 rows/sec., 68.00 B (1.79 MiB/sec.), compressed 0.4594594594594595 times to 148.00 B (3.88 MiB/sec.)
2023.10.20 11:00:00.829544 [ 154098 ] {b4a2c158-c3f3-4469-9372-7837ba549e13} <Debug> Connection (127.0.0.3:9000): Sent data for 1 external tables, total 1 rows in 7.3551e-05 sec., 13468 rows/sec., 36.00 B (472.54 KiB/sec.), compressed 0.3673469387755102 times to 98.00 B (1.25 MiB/sec.)
2023.10.20 11:00:00.829798 [ 5675 ] {af251524-3437-446b-bf8f-f927cf35bd84} <Debug> executeQuery: (from [::ffff:127.0.0.1]:33922, initial_query_id: b4a2c158-c3f3-4469-9372-7837ba549e13) (comment: 01290_max_execution_speed_distributed.sql) SELECT count() FROM numbers(1000000) (stage: WithMergeableState)
2023.10.20 11:00:01.672817 [ 154098 ] {b4a2c158-c3f3-4469-9372-7837ba549e13} <Trace> StorageDistributed (remote): (127.0.0.2:9000) Cancelling query
2023.10.20 11:00:01.865918 [ 5057 ] {b4a2c158-c3f3-4469-9372-7837ba549e13} <Error> executeQuery: Code: 49. DB::Exception: Received from 127.0.0.3:9000. DB::Exception: Bad cast from type DB::ColumnVector<unsigned long> to DB::ColumnAggregateFunction. Stack trace:
2023.10.20 11:00:01.869475 [ 5057 ] {b4a2c158-c3f3-4469-9372-7837ba549e13} <Error> TCPHandler: Code: 49. DB::Exception: Received from 127.0.0.3:9000. DB::Exception: Bad cast from type DB::ColumnVector<unsigned long> to DB::ColumnAggregateFunction. Stack trace:
2023.10.20 11:00:01.870487 [ 5057 ] {b4a2c158-c3f3-4469-9372-7837ba549e13} <Debug> TCPHandler: Processed in 1.181866328 sec.
2023.10.20 11:00:01.870552 [ 5057 ] {b4a2c158-c3f3-4469-9372-7837ba549e13} <Debug> MemoryTracker: Peak memory usage (for query): 7.03 MiB.
```

There must be some setting changing the behaviour because this query comes straight from a test that passes normally in CI.

### PR DESCRIPTION
**Pull request by antonio2368**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Disable experimental partial results for secondary queries. Partial results could produce data of invalid type when secondary queries contained aggregation.

cc @vdimir I saw you're familiar with this feature so you can probabl verify if it's okay

Fix https://github.com/ClickHouse/ClickHouse/issues/55864

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 3d50b4cd99de8edbb714c8e66f531ee58199f0da with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/55870/3d50b4cd99de8edbb714c8e66f531ee58199f0da/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6587796004">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/55870/3d50b4cd99de8edbb714c8e66f531ee58199f0da/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Interpreters/InterpreterSelectQuery.cpp [hunk 1]
```diff
@@ -3167,16 +3167,23 @@ void InterpreterSelectQuery::initSettings()
...
+    if (client_info.query_kind == ClientInfo::QueryKind::SECONDARY_QUERY)
     {
-        /// Disable two-level aggregation due to version incompatibility.
-        context->setSetting("group_by_two_level_threshold", Field(0));
-        context->setSetting("group_by_two_level_threshold_bytes", Field(0));
+        /// We can try to send invalid data type to the initiator when sending partial results
+        /// There can be other additional problems related to sending partial results to the initator node
+        /// and there is no other useful usage of the partial results that is not on the client side
+        /// so we are always disabling it for secondary queries
+        //context->setSetting("partial_result_update_duration_ms", Field(0));
```
Comment (vdimir) [inline src/Interpreters/InterpreterSelectQuery.cpp:3179]:
```suggestion
        context->setSetting("partial_result_update_duration_ms", Field(0));
```

Comment (antonio2368) [reply src/Interpreters/InterpreterSelectQuery.cpp:3179]: yes, was verifying the fix...
Thanks!

**Review by vdimir — APPROVED** [review]
So the idea is that only initial query should produce partial_results? Sounds reasonable, we fetch data from remote server, but for emitting partial result we use internal aggregation/sorting/etc state and it's not clear how to merge partial results from several servers.

File: src/Interpreters/InterpreterSelectQuery.cpp (hunk 1 above)
Comment (vdimir) [reply src/Interpreters/InterpreterSelectQuery.cpp]: I need to fix `TestsBugfixCheck` for that

**Comment by antonio2368** [comment]
> So the idea is that only initial query should produce partial_results?

Yes.
Also, it's not clear if partial results will even work on the initiator if we disable it for secondary queries.
But either way the feature is experimental, works for `max_threads=1` only so it should be disabled for distributed queries also until it's properly tested and implemented.

**Pull request closed without merging**